### ISSUE TITLE
Installation Error with mamba-ssm while Installing nemo-curator

### ISSUE THREAD

**Issue by piyushkashyap07** [NVIDIA/NeMo-Curator#399]
**Description:**

I'm encountering an issue while attempting to install `nemo-curator[all]` via the command in google colab:

```bash
!pip install --extra-index-url https://pypi.nvidia.com nemo-curator[all]
```

The error log indicates that the build process for the `mamba-ssm` package fails, resulting in the following message:

```
ERROR: Failed to build installable wheels for some pyproject.toml based projects (mamba-ssm)
```

**Error Details:**

- The installation completes for some packages but fails when building the wheel for `mamba-ssm`.
- The error originates from a subprocess, and it does not seem to be a pip issue.
  
**Solution:**

I was able to resolve the issue by manually downloading and installing the specific `.whl` file for `mamba-ssm`. Below are the working commands:

```bash
!wget https://github.com/state-spaces/mamba/releases/download/v2.2.2/mamba_ssm-2.2.2+cu118torch2.1cxx11abiFALSE-cp310-cp310-linux_x86_64.whl
!pip install mamba_ssm-2.2.2+cu118torch2.1cxx11abiFALSE-cp310-cp310-linux_x86_64.whl
```

These commands allow for a successful installation of the `mamba-ssm` package.

**Environment:**

- Python version: 3.10
- OS: Linux (x86_64)
- CUDA version: cu118 (for torch 2.1)

**Additional Information:**

Would appreciate any updates or recommendations regarding the installation process for `mamba-ssm` in the `nemo-curator` package. This solution might be useful for others facing the same issue.

**Comment by Eom-taeseon** [comment]
I have same error

**Comment by piyushkashyap07** [comment]
try this 
!wget https://github.com/state-spaces/mamba/releases/download/v2.2.2/mamba_ssm-2.2.2+cu118torch2.1cxx11abiFALSE-cp310-cp310-linux_x86_64.whl
!pip install mamba_ssm-2.2.2+cu118torch2.1cxx11abiFALSE-cp310-cp310-linux_x86_64.whl

before 
!pip install --extra-index-url https://pypi.nvidia.com nemo-curator[all]


will work .

**Comment by ayushdg** [comment]
Thanks for raising @piyushkashyap07. @ryantwolf looks like this is related to Nemo-toolkit 2.0 which was fixed in #372 but still impacts the existing release. Perhaps we need to patch release `0.5.1` pinning nemo-toolkit<2?

**Comment by sarahyurick** [comment]
Hi, should now be resolved with the latest NeMo Curator release 0.5.1. Thanks!
